addons/account/models/account_invoice.py

```diff
@@ -1693,6 +1635,35 @@ def _prepare_refund(self, invoice, date_invoice=None, date=None, description=Non
             values['date'] = date
         if description:
             values['name'] = description
+
+        # Treat refund tax lines. They should match the incoice's, except for the data
+        # directly coming from repartition_lines, which should be replaced by their
+        # refund equivalent
+        tax_rep_ln_mapping = {}
+        for invoice_tax_entry in invoice.tax_line_ids:
+            if not invoice_tax_entry.tax_repartition_line_id.tax_id in tax_rep_ln_mapping:
```

Comment:
tax_id as key ? something is escaping my understanding

Reply:
Oops. Indeed, it worked by chance (and did useless operations).